src/umd/shims/ReactAddonsDOMDependenciesUMDShim.js

```diff
@@ -9,16 +9,12 @@
  * @providesModule ReactAddonsDOMDependenciesUMDShim
  */
 
-/* globals ReactDOM */
-
 'use strict';
 
 exports.getReactDOM = function() {
-  return ReactDOM;
-};
-
-exports.getReactInstanceMap = function() {
-  return ReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactInstanceMap;
+  var ReactWithAddonsUMDEntry = require('ReactWithAddonsUMDEntry');
+  // This is injected by the ReactDOM UMD build:
+  return ReactWithAddonsUMDEntry.__SECRET_INJECTED_REACT_DOM_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
```

Comment:
The worst case (it doesn't exist) is the same worst case we already had before when `ReactDOM` wasn't available yet, but at least this works with AMD. I don't think it's realistic that people will get here though, unless they have mismatching React versions (which breaks in other subtle ways anyway).